### PR TITLE
Add lab directories and READMEs

### LINKED ISSUE #57: Create lab directories on go-training
`NN_lab_name` for : 
- Lab 1 - Fibonacci
- Lab 2 - Bubble sort
- Lab 3 - Letter frequency
- Lab 4 - Numeronym
- Lab 5 - Stringer
- Lab 6 - CRUD puppy with interface
- Lab 7 - Errors
- Lab 8 - Project Layout
- Lab 9 - JSON puppy
- Lab 10 - Puppy REST
- Lab 11 - Puppy Notifications

### PR DESCRIPTION
**Pull request by anzboi**
Adds a directory and README for each lab. directory names and
README's taken from the collated lab assignment gist.

I hope I got the names right.
Good opportunity to have another look at the lab instructions too.

Fixes [#57](https://github.com/anz-bank/go-slides/issues/57)

### CODE REVIEW

File: 01_fibonacci/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Lab 1 - Fibonacci
+
+Create an executable go program in directory 01_fib/USERNAME
```
Comment (camscale) [inline 01_fibonacci/README.md]: this says `01_fib` but the readme is in `01_fibonacci`. please fix one or the other.

File: 02_bubble_sort/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Lab 2 - Bubble sort
+
+Create an executable go program in directory `02_bubble/USERNAME`
```
Comment (camscale) [inline 02_bubble_sort/README.md]: ditto

File: 03_letter_frequency/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Lab 3 - Letter frequency
+
+Create an executable go program in directory `03_letters/USERNAME`
```
Comment (camscale) [inline 03_letter_frequency/README.md]: ditto

File: 06_CRUD_puppy_with_interface/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Lab 6 - CRUD puppy with interface
+
+Create an executable go program in directory `06_puppy/USERNAME` (see [hints]([https://github.com/anz-bank/go-samplerest/blob/master/pkg/pet/types.go))
```
Comment (camscale) [inline 06_CRUD_puppy_with_interface/README.md]: ditto

File: 07_errors/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Lab 7 - Errors
+
+Create an executable go program in directory `07_errors/USERNAME`
+Copy the CRUD puppy from upstream master `06_puppy/USERNAME`
```
Comment (camscale) [inline 07_errors/README.md:4]: ditto

File: 08_project_layout/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Lab 8 - Project Layout
+
+Copy the CRUD puppy from upstream master `07_errors/USERNAME`
+Create directory `08_project/USERNAME` containing
```
Comment (camscale) [inline 08_project_layout/README.md]: ditto

File: 09_JSON_puppy/README.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# Lab 9 - JSON puppy
+
+Create directory `09_json/USERNAME` containing a copy of upstream master `08_project/USERNAME`
```
Comment (camscale) [inline 09_JSON_puppy/README.md]: ditto

File: 10_puppy_REST/README.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# Lab 10 - Puppy REST
+
+Create directory `10_rest/USERNAME` containing a copy of upstream master `09_json/USERNAME`
```
Comment (camscale) [inline 10_puppy_REST/README.md]: ditto

File: 11_puppy_notifications/README.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Lab 11 - Puppy Notifications
+
+Create directory `11_notify/USERNAME` containing a copy of upstream master `10_rest/USERNAME`
```
Comment (camscale) [inline 11_puppy_notifications/README.md]: ditto

File: 01_fibonacci/README.md (hunk 1 above)
Comment (anzboi) [reply 01_fibonacci/README.md]: Oh, thats where they're located

File: 07_errors/README.md (hunk 5 above)
Comment (anzboi) [reply 07_errors/README.md:4]: This one is actually ok, directory to create is `07_errors/USERNAME`

Comment (camscale) [reply 07_errors/README.md:4]: My comment was on the `06_puppy` line, which did not match.

**Review by camscale — APPROVED** [review]
LGTM

**Pull request merged**